src/frontend/src/components/RecordSearch/Record/RecordSummary/CaseBalances.tsx

```diff
@@ -1,47 +1,43 @@
 import React from "react";
+import { useSelector } from "react-redux";
 import BalanceList from "./BalanceList";
 import { CaseData } from "../types";
+import { AppState } from "../../../../redux/store";
+import { selectCasesWithBalanceDue } from "../../../../redux/search/selectors";
 import currencyFormat from "../../../../service/currency-format";
 
 interface Props {
-  cases?: CaseData[];
   totalBalance: number;
 }
 
-const hasBalanceDue = (element: CaseData) => {
-  return element.balance_due > 0;
-};
-
-export default class CaseBalances extends React.Component<Props> {
-  render() {
-    if (!this.props.cases) {
-      return null;
-    }
-
-    const listItems = this.props.cases
-      .filter(hasBalanceDue)
-      .map((element: CaseData) => {
-        return (
-          <li className="mb2" key={element.case_number}>
-            <a
-              className="link underline hover-blue break-all"
-              href={`#${element.case_number}`}
-            >
-              {element.case_number}
-            </a>{" "}
-            <span className="fr">{currencyFormat(element.balance_due)}</span>
-          </li>
-        );
-      });
+const CaseBalances: React.FunctionComponent<Props> = (props) => {
+  const cases = useSelector((state: AppState) => {
+    return selectCasesWithBalanceDue(state);
+  });
 
+  const listItems = cases.map((element: CaseData, i: number) => {
     return (
-      <div className="mb4">
-        <BalanceList
-          listItems={listItems}
-          title="Balance Due by Case"
-          totalBalance={this.props.totalBalance}
-        />
-      </div>
+      <li className="mb2" key={i}>
```

Comment:
I think this key should be `case_number`.

Reply:
This has been updated